src/MultiRegion/cubed_sphere_boundary_conditions.jl

```diff
@@ -39,22 +20,23 @@ function fill_halo_regions!(field::CubedSphereField{<:Center, <:Center})
     kernel_parameters = KernelParameters((Hc, Nc, Nz), (0, 0, 0))
     @apply_regionally begin
         launch!(grid.architecture, grid, kernel_parameters, _fill_cubed_sphere_center_center_field_east_west_halo_regions!,
-                field, multiregion_field, n_regions, region, Hc, Nc)
+                field, multiregion_field, region, grid.connectivity.connections, Hc, Nc)
     end
 
     kernel_parameters = KernelParameters((Nc, Hc, Nz), (0, 0, 0))
     @apply_regionally begin
         launch!(grid.architecture, grid, kernel_parameters, _fill_cubed_sphere_center_center_field_north_south_halo_regions!,
-                field, multiregion_field, n_regions, region, Nc, Hc)
+                field, multiregion_field, region, grid.connectivity.connections, Nc, Hc)
```

Comment:
I don't see why we need to give `grid.architecture`, `grid`, and `grid.connectivity.connections` as separate args. Why not just provide `grid` as argument and then `grid.architecture` and `grid.connectivity.connections` are all included?

Reply:
oh now I see: `launch!` needs `(arch, grid, params, kernel, args...)`...
so OK, I get why `grid.architecture is first arg; but why do we need to provide `grid.connectivity.connections` as separate argument?

Reply:
probably no actually, because `getregion` happens on the arguments of `launch!`, not before

Reply:
gotcha!

Reply:
Correct. Since `@apply_regionally` happens on the arguments of `launch!`, I end up passing `grid.connectivity.connections[region]` for each `region`, which is defined only for `ConformalCubedSphereGrid`, and not for `OrthogonalSphericalShellGrid`.

Reply:
Even though `test_multiregion_cubed_sphere` runs smoothly on the CPU, I do get multiple errors on the GPU. 🙁

Reply:
looks like the problem is solved. There is a out-of-bound access in the metrics though

Reply:
btw, if we are gonna call `fill_halo_regions!` with `@apply_regionally` in front, then how will the methods with typed args as `CubedSphereField`s, e.g., 

```
fill_halo_regions!(field::CubedSphereField{<:Face, <:Face})
```

ever are used?

Reply:
`fill_halo_regions!` is not a local function because it requires communication so it should not be called with `@apply_regionally` but it should contain apply regionally inside